freud/diffraction.pyx

```diff
@@ -21,13 +21,225 @@ import scipy.ndimage
 import freud.locality
 
 cimport numpy as np
+from cython.operator cimport dereference
 from libcpp cimport bool as cbool
 
+cimport freud._diffraction
+cimport freud.locality
 cimport freud.util
-from freud.util cimport _Compute
+from freud.locality cimport _SpatialHistogram1D
+from freud.util cimport _Compute, vec3
 
 logger = logging.getLogger(__name__)
 
+cdef class StaticStructureFactorDebye(_Compute):
+    R"""Computes a 1D static structure factor.
+
+    This computes the static `structure factor
+    <https://en.wikipedia.org/wiki/Structure_factor>`__ :math:`S(k)`, assuming
+    an isotropic system (averaging over all :math:`k` vectors of the same
+    magnitude). This is implemented using the Debye scattering equation:
+
+    .. math::
+
+        S(k) = \frac{1}{N} \sum_{i=0}^{N} \sum_{j=0}^{N} \text{sinc}(k r_{ij})
+
+    where :math:`N` is the number of particles, :math:`\text{sinc}` function is
+    defined as :math:`\sin x / x` (no factor of :math:`\pi` as in some
+    conventions). For more information see `here
+    <https://en.wikipedia.org/wiki/Structure_factor>`__. The equation 4 from
+    the link can be obtained by replacing :math:`\frac{\sin(k r)}{kr}` with
+    :math:`\text{sinc}(k r)`.
+
+    .. note::
+        This code assumes all particles have a form factor :math:`f` of 1.
+
+    Partial structure factors can be computed by providing ``query_points`` to the
+    :py:meth:`compute` method. When computing a partial structure factor, the
+    total number of points in the system must be specified. The normalization
+    criterion is based on the Faber-Ziman formalism. For particle types
+    :math:`\alpha` and :math:`\beta`, we compute the total scattering function
+    as a sum of the partial scattering functions as:
+
+    .. math::
+
+        S(k) - 1 = \sum_{\alpha}\sum_{\beta} \frac{N_{\alpha}
+        N_{\beta}}{N_{total}^2} \left(S_{\alpha \beta}(k) - 1\right)
+
+    This class is based on the MIT licensed `scattering library
+    <https://github.com/mattwthompson/scattering/>`__ and literature references
+    :cite:`Liu2016`.
```

Comment:
Additional suggested citations for Faber-Ziman:
- https://openscholarship.wustl.edu/etd/1358/
- http://isaacs.sourceforge.net/manual/page26_mn.html
- https://www.tandfonline.com/doi/abs/10.1080/14786436508211931

Reply:
@DomFijan suggests adding code examples, especially for cross-term scattering -- we can use the tests as a reference for this.